### DISCUSSION TITLE
MQTT Driven Tiles

### DISCUSSION THREAD

**Discussion by gary-hammer** [Daveiano/weewx-wdc#96] [Ideas]
Any chance of using MQTT data to update tiles?
Or, to auto refresh the page on an interval?

I'd like to also display this on a tablet used as a weather console. Without a way to have new data shown, I can't.

**Comment by Daveiano** [comment]
Support for Live updates via MQTT is planned for the skin, but will not hit anytime soon I guess.

Auto refresh would be easy to implement: you could try this by adding 

`<meta http-equiv="refresh" content="10">`  (10 is the refresh interval in seconds)

to the `includes/html-head.inc` template (for testing right at the top). Please let me know if this works for you, so I would add this to the skin and make the refresh interval configurable via skin.conf.

**Comment by gary-hammer** [comment]
I gave the refresh a try and after a day or so, stopped.
Refresh timed to the browser, according to when in the cycle you launch the page, may launch anywhere in the 5 minute window so could be almost 10 minutes out of sync. We use the tablet display mostly when it is stormy, either winter or summer. Then, wind and rain are what we look for.
A quick glance as your are heading out the door to see the temp, refresh method is fine.

I'll just wait patiently for the real time capability.

**Comment by Daveiano** [comment]
Thank you for being so patient!

Another thing I am thinking of when using the refresh interval could be this: https://www.weewx.com/docs/customizing.htm#customizing_gen_time

You could possibly tell weewx to generate the report every 2 mins or so and just set the refresh interval to 1 min. This way the page should be relatively up-to-date.

However, this will hit in a later version, like the MQTT implementation.

**Reply by gary-hammer** [reply]
No problem, while I'm looking forward to your new features, it's excellent now.

I don't believe this will work without changing the archive interval. There's a note right before report_timing.

`Note
While report_timing specifies when a given report should be generated, the generation of reports is still controlled by the WeeWX report cycle, so reports can never be generated more frequently than once every archive period.`

**Reply by Daveiano** [reply]
You are right, this will not work, I totally overlooked this.

**Comment by Daveiano** [comment]
@gary-hammer I am currently testing Live updates via MQTT on my personal homepage for the stat tiles on the index page: https://www.weewx-hbt.de/

Sadly my station only supports sending data every 40s, so it's a bit lame...

This will hit soon in a new release :)

**Reply by gary-hammer** [reply]
Yes!
Can't wait!
